### DISCUSSION TITLE
Allow modification of the header actions menu

### DISCUSSION THREAD

**Discussion by BennyAlex** [KevinVandy/material-react-table#566] [Ideas]
Hello,

i would like to customize the actions menu (when clicking the 3 dots on the right of the header name).
For example I dont want do display certain options there or maybe add custom ones.

It also doesnt make sense to display the filter by 'name' option (in a disabled state) at all, when the showColumnFilters state is true. Its just confusing for the user.

**Comment by joey-kwl** [comment]
You can make your own [action menu]( https://www.material-react-table.com/docs/guides/column-actions#custom-column-actions-menu), but I don't know how to disable/hide the original items.

**Comment by BennyAlex** [comment]
@KevinVandy I think it would be a nice enhancement to block certain entries from the default menu

**Reply by KevinVandy** [reply]
This should be fixed in v1.14.0 with the internal `internalColumnMenuItems` being an optional inclusion.

https://www.material-react-table.com/docs/guides/column-actions#custom-column-actions-menu
